lib/omninav/builder.rb

```diff
@@ -0,0 +1,641 @@
+        logo_html: build_logo,
+        search_html: build_search,
+        login_html: build_login,
+        off_canvas_nav_html: build_off_canvas_nav
+      }
+
+      # rubocop:disable Rails/OutputSafety
+      # App controls this string so we know it's safe. Is there better way to do this?
+      format(navbar_template, sections).html_safe
+      # rubocop:enable Rails/OutputSafety
+    end
+
+    def build_version
+      format('%s.%s', VERSION, Time.zone.now.strftime('%Y%m%d.%H%M%S'))
+    end
+
+    def prep_build
+      # Prep staging/output directories.
+      [@staging_dir, @output_dir].each do |dir|
+        FileUtils.rm_rf dir
+        FileUtils.mkdir_p dir
+      end
+    end
+
+    def generate_markup_file
+      markup_file_path = @staging_dir.join(@markup_file)
+      omninav_html = build_html
+      File.open(markup_file_path, 'w') { |file| file.write(omninav_html) }
+    end
+
+    def move_output_files_to_build_directory
+      @deploy_map.each do |staging_name, deploy_name|
+        staging_file = @staging_dir.join(staging_name)
+        deploy_file = @output_dir.join(deploy_name)
+
+        if staging_file.to_s.include? '*'
+          Dir.glob(staging_file).each do |path|
+            FileUtils.mv path, deploy_file
+            puts format('Writing %s to %s.', path, deploy_file)
+          end
+        else
+          FileUtils.mv staging_file, deploy_file
+          puts format('Writing %s to %s.', staging_file, deploy_file)
+        end
+      end
+    end
+
+    def cleanup
+      FileUtils.rm_rf @staging_dir
+    end
+
+    private
+
+    #
+    # Markup Methods
+    #
+    # Note: when updating navbar markup, you will most likely be changing one of these
+    # methods.
+    #
+    # rubocop:disable Metrics/MethodLength
+    def navbar_template
+      <<-NAVBAR_HTML
+%<header>s
+
+<!-- OmniNav NavBar -->
+<div id="cu_nav" class="omninav-builder use-transitions">
+
+%<logo_html>s
+%<search_html>s
+%<login_html>s
+%<off_canvas_nav_html>s
+
+</div>
+<!-- End OmniNav NavBar -->
+
+NAVBAR_HTML
+    end
+
+    def build_header
+      if @target == 'blogs'
+        header_for_blogs
+      else
+        format('<!-- OmniNav Build Version: %s -->', build_version)
+      end
+    end
+
+    def header_for_blogs
+      template = <<-HEADER_PHP
+<?php
+// OmniNav Build Version: %<version>s
+
+// If logged in
+if (is_user_logged_in()) {
+
+    global $current_user;
+    get_currentuserinfo();
+}
+?>
+HEADER_PHP
+
+      params = {
+        version: build_version
+      }
+      format(template, params)
+    end
+
+    def build_logo
+      template = <<-LOGO_HTML
+  <!-- Logo Block -->
+  <a class="cu-logo" href="%<domain>s" title="Chapman University Website Home Page">
+    %<logo>s
+  </a>
+LOGO_HTML
+
+      params = {
+        domain: DOMAIN,
+        logo: SvgImage.chapman_horizontal_color_logo
+      }
+      format(template, params)
+    end
+
+    def build_search
+      template = <<-SEARCH_HTML
```

Comment:
So say you need to modify the search form, this is where you would edit the markup. You would still edit the stylesheets to change styling. And you could quickly review changes in the new omninav controller.